src/Permission.java

```diff
@@ -0,0 +1,54 @@
+import java.util.UUID;
+
+/**
+ * the default implementation of permission, that users has and used to invoke
+ * command this is created for verification purpose.
+ */
+public final class Permission implements PermissionSecondary {
```

Comment:
Naming wise, this should be called something like `Permission1`. It correctly inherits from `PermissionSecondary`, but `PermissionSecondary` should be an abstract class inheriting from `Permission`.